tests/sweep_framework/sweeps/all_gather_T3K_sharded.py

```diff
@@ -0,0 +1,239 @@
+# SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Optional, Tuple
+
+import torch
+
+import ttnn
+from models.utility_functions import get_devices_for_t3000
+from tests.ttnn.utils_for_testing import check_with_pcc, start_measuring_time, stop_measuring_time
+from loguru import logger
+import tt_lib as ttl
+import ttnn
+from tests.tt_eager.python_api_testing.sweep_tests.comparison_funcs import comp_equal, comp_pcc
+from tests.ttnn.unit_tests.operations.test_all_gather import is_unsupported_case
+from ttnn import ShardTensorToMesh
+
+# Override the default timeout in seconds for hang detection.
+TIMEOUT = 30
+
+# Parameters provided to the test vector generator are defined here.
+# They are defined as dict-type suites that contain the arguments to the run function as keys, and lists of possible inputs as values.
+# Each suite has a key name (in this case "suite_1" and "suite_2") which will associate the test vectors to this specific suite of inputs.
+# Developers can create their own generator functions and pass them to the parameters as inputs.
+parameters = {
+    "all_gather": {
+        "num_devices": [4, 8],
+        "num_links": [1, 2],
+        "input_shape": [
+            # [1, 1, 32, 32], # throwing python segmentation fault
+            [1, 1, 32, 1024],
+            [4, 1, 32, 128],
+            [8, 1, 32, 1024],
+            [4, 1, 32, 32],
+            [1, 8, 32, 1024],
+        ],
+        "input_shard_shape": [
+            [32, 1024],
+            [32, 128],
+            [32, 32],
+        ],
+        "shard_grid": [
+            ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(7, 3))}),
```

Comment:
We should be sweeping over many more shard grid sizes (and offsets). These are only 2 shard grids that are very "friendly". We need to include the other shard grid sizes too (those with offsets and those with varying widths/heights). At the very least we need non-power-of-2 and non-multiple-of-2 sizes included.

Additionally, atleast one prime value size per dim should be included.

These are _sweeps_ so we should essentially be able to enumerate all the base cases.